src/types/subgraph/codegen.ts

```diff
@@ -0,0 +1,23 @@
+import { CodegenConfig } from '@graphql-codegen/cli';
+
+const config: CodegenConfig = {
+  schema: process.env.SUBGRAPH_GRAPHQL_ENDPOINT,
+  documents: ['src/clients/subgraph/**/*.graphql'],
+  generates: {
+    './src/types/subgraph/gql/queries.ts': {
+      plugins: [
+        {
+          add: {
+            placement: 'append',
+            content: `export const subgraphEndpoint = '${process.env.SUBGRAPH_GRAPHQL_ENDPOINT}';`,
```

Comment:
Following my previous comment, the subgraph endpoint would be imported from the config:
```suggestion
            content: `export const subgraphEndpoint = '${config.subgraphEndpointUrl}';`,
```

Reply:
I think I have to keep this as it was, because if I try to import config in the codegen file, it won't understand the imports inside `config/index.ts`:

```
import { BscChainId } from 'types';
import { BSC_SCAN_URLS } from 'constants/bsc';
import { API_ENDPOINT_URLS, RPC_URLS } from 'constants/endpoints';
```

Because those are aliased imports as set in tsconfig, and the CLI needs relative paths. Going down this route means changing all the related imports to be like:

```
import { BscChainId } from '../types';
import { BSC_SCAN_URLS } from '../constants/bsc';
import { API_ENDPOINT_URLS, RPC_URLS } from '../constants/endpoints';
```
I think it's just simpler to keep the script and the env var, what do you think?